### PR TITLE
DataDefinitions: Ship: simplify Ship.SpokenName()

### PR DESCRIPTION
**Pull request by intelfx**
This is a trivial refactoring for code clarity with no behavior changes.

### CODE REVIEW

**Review by Tkael — APPROVED** [review]
We could potentially further improve by removing `string result;` and returning the results more directly but this is still good as-is.

File: DataDefinitions/Ship.cs [hunk 1]
```diff
@@ -411,8 +411,7 @@ public override string ToString()
 
         public string SpokenName(string defaultname = null)
         {
-            string ship = (defaultname ?? phoneticmodel) ?? Properties.Ship._ship;
-            string result = Properties.Ship.your + " " + ship;
+            string result;
```
Comment (GioVAX) [inline DataDefinitions/Ship.cs:414]: We can remove `result` altogether, and just `return` the computed values.

**Review by GioVAX — CHANGES_REQUESTED** [review]

File: DataDefinitions/Ship.cs (hunk 1 above)
Comment (intelfx) [reply DataDefinitions/Ship.cs:414]: Done.

**Comment by richardbuckle** [comment]
> We could potentially further improve by removing `string result;` and returning the results more directly but this is still good as-is.

I wouldn't call that an improvement. Having a result variable lets you inspect it in the debugger and the optimiser will elide it for the release target.

**Comment by intelfx** [comment]
So, what's the verdict? I don't care, I can rollback that change or not.

**Comment by intelfx** [comment]
I have force-pushed the original version, with an explicit `string result`.

**Pull request merged**